src/models/room.ts

```diff
@@ -914,12 +914,38 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.myUserId;
     }
 
+    // gets service members (e.g. helper bots) for exclusion
+    private getFunctionalMembers(): string[] {
+        const mFunctionalMembers = this.currentState.getStateEvents(UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name, "");
+        if (Array.isArray(mFunctionalMembers?.getContent().service_members)) {
+            return mFunctionalMembers!.getContent().service_members;
+        }
+        return [];
+    }
+
+    private getInvitedAndJoinedFunctionalMemberCount(functionalMembers: String[]): number {
+        return functionalMembers.reduce((count, m) => {
+            const membership = this.getMembers().find((member) => member.userId === m)?.membership;
```

Comment:
calling `getMembers` and searching through it for each entry in `functionalMembers` seems rather inefficient, particularly if the list of `functionalMembers` is long. Please can you find a better way to do this? (Maybe iterate through `getMembers` looking for entries that are in `functionalMembers`, rather than the other way around?)